src/seedu/addressbook/data/person/address/Street.java

```diff
@@ -0,0 +1,51 @@
+package seedu.addressbook.data.person.address;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's street in a Person's Address the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidStreetAddress(String)}
+ */
+public class Street {
+    public static final String EXAMPLE = "Clementi Ave 3";
+    public static final String MESSAGE_ADDRESS_STREET_CONSTRAINTS = "Address Street can be in any format";
+    public static final String ADDRESS_STREET_VALIDATION_REGEX = ".+";
+
+    public final String value;
```

Comment:
`value` should be declared `private` for better encapsulation. Also would be better if you change the name instead of using `value` which is to general, maybe `_street`.